frontend/src/Login.tsx

```diff
@@ -1,22 +1,70 @@
-import React, { useState } from "react";
+import React, { FormEvent, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import Container from "react-bootstrap/Container";
 import Form from "react-bootstrap/Form";
 import Button from "react-bootstrap/Button";
 import Alert from "react-bootstrap/Alert";
 import Stack from "react-bootstrap/Stack";
+import Row from "react-bootstrap/Row";
 
+import { userContext } from "./Context/UserContext";
+
+import { useCookies } from "react-cookie";
 function Login() {
   const [error, setError] = useState("");
+  const [cookies, setCookies] = useCookies(["jwt"]);
+
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [user, setUser] = useState({ user: { token: "" } });
+  const [token, setToken] = useState("");
+
   const nav = useNavigate();
 
-  const submit: React.FormEventHandler<HTMLFormElement> = (
-    e: React.FormEvent<HTMLFormElement>
-  ) => {
+  async function handleLogin(username: string, password: string) {
+    return fetch("http://localhost:8080/login", {
+      method: "POST",
+      mode: "cors",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        username: username,
+        password: password,
+      }),
+    })
+      .then((response) => response.json())
+      .then((response) => {
+        //TODO: IDK WHY BUT it doesn't seem to catch 401 unauthorized. pls help
+        setCookies("jwt", response.token);
+      })
```

Comment:
```suggestion
      .then((response) => {
        if (response.status == 401)
          throw 'Unauthorized';
        return response.json();
      })
      .then((response) => {
        setCookies("jwt", response.token);
      })
```

Reply:
Docs for this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw